ext/crypto/00_crypto.js

```diff
@@ -2464,27 +2465,52 @@
         let hash;
 
         // 8.
-        switch (jwk.alg) {
-          case undefined:
-            hash = undefined;
-            break;
-          case "RS1":
-            hash = "SHA-1";
-            break;
-          case "RS256":
-            hash = "SHA-256";
-            break;
-          case "RS384":
-            hash = "SHA-384";
-            break;
-          case "RS512":
-            hash = "SHA-512";
-            break;
-          default:
-            throw new DOMException(
-              `'alg' property of JsonWebKey must be one of 'RS1', 'RS256', 'RS384', 'RS512'`,
-              "DataError",
-            );
+        if (normalizedAlgorithm.name === "RSA-OAEP") {
+          switch (jwk.alg) {
+            case undefined:
+              hash = undefined;
+              break;
+            case "RSA-OAEP":
+              hash = "SHA-1";
+              break;
+            case "RSA-OAEP-256":
+              hash = "SHA-256";
+              break;
+            case "RSA-OAEP-384":
+              hash = "SHA-384";
+              break;
+            case "RSA-OAEP-512":
+              hash = "SHA-512";
+              break;
+            default:
+              throw new DOMException(
+                `'alg' property of JsonWebKey must be one of 'RSA-OAEP', 'RSA-OAEP-256', 'RSA-OAEP-384', or 'RSA-OAEP-512'`,
+                "DataError",
+              );
+          }
+        } else {
```

Comment:
@lucacasonato missing `RSA-PSS` and switch-case `PS256` etc

Reply:
Oh darn, I thought RSA-PSS and RSASSA-PKCS1-v1_5 used the same algorithm names... my bad! I have an open issue to add WPTs for this: https://github.com/w3c/webcrypto/issues/304.

I'll go adapt your tests for the time being (also need to add the types for JWK exports still).